apps/s_client.c

```diff
@@ -2142,6 +2144,37 @@ int s_client_main(int argc, char **argv)
                 goto shut;
         }
         break;
+    case PROTO_SIEVE:
+        {
+            int foundit = 0;
+            BIO *fbio = BIO_new(BIO_f_buffer());
+            BIO_push(fbio, sbio);
+            /* wait for multi-line response to end from Sieve */
+            do {
+                mbuf_len = BIO_gets(fbio, mbuf, BUFSIZZ);
+                /*
+                 * According to RFC 5804 § 1.7, capability
+                 * is case-insensitive, make it uppercase
+                 */
+                if (mbuf_len > 1 && mbuf[0] == '"') {
+                    for (i = 0; mbuf[i] != '\0'; i++)
```

Comment:
There's a similar loop in apps/ca.c  Create a make_uppercase utility routine in apps/apps.[ch] and use that there and here.